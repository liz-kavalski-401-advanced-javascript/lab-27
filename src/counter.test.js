import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';


import Counter from '../src/components/counter/counter.js';

describe('<Counter/>', ()=> {
  it('renders without crashing', () => {
    const component = shallow(<Counter/>);
    expect(component.find('section').exists()).toBeTruthy()
  });

  it('changes state on a click to less then 0',() => {
    let component = mount(<Counter/>);
    let button = component.find(".downclicker");
    button.simulate('click');
    
    expect(component.state('count')).toBeLessThanOrEqual(0);
  });

  it('changes state on a click to greater then 0',() => {
    let component = mount(<Counter/>);
    let button = component.find(".upclicker");
    button.simulate('click');
    
    expect(component.state('count')).toBeGreaterThanOrEqual(0);
  });


  it('renders correctly', () =>{
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
