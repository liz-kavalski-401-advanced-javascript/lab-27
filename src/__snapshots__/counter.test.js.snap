// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> renders correctly 1`] = `
<section
  className="counter"
>
  <a
    className="downclicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="upclicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
